corehq/apps/change_feed/pillow.py

```diff
@@ -9,16 +9,18 @@
 from corehq.apps.change_feed.topics import get_topic
 from couchforms.models import all_known_formlike_doc_types
 import logging
+from pillowtop.checkpoints.manager import PillowCheckpoint, get_django_checkpoint_store
 from pillowtop.couchdb import CachedCouchDB
 from pillowtop.listener import PythonPillow
 
 
 class ChangeFeedPillow(PythonPillow):
 
-    def __init__(self, couch_db, kafka):
+    def __init__(self, couch_db, kafka, checkpoint):
         super(ChangeFeedPillow, self).__init__(couch_db=couch_db)
         self._kafka = kafka
         self._producer = KeyedProducer(self._kafka)
+        self._checkpoint = checkpoint
```

Comment:
yeah not the final design. i'm optimistic by the time i'm done it will be very clear (and then i can add docs). i can update my brain dump on current thinking somewhere if you think that's useful, though am figuring a bit out as i go and expecting it to be a little messy in the intermediate stages